includes/class-magic-link.php

```diff
@@ -379,6 +407,19 @@ public static function generate_token( $user ) {
 		return $token_data;
 	}
 
+	/**
+	 * Generate secret.
+	 *
+	 * @param \WP_User $user User to generate the secret for.
+	 *
+	 * @return string
+	 */
+	public static function generate_secret( $user ) {
+		$secret = str_replace( '-', '', wp_generate_uuid4() );
```

Comment:
Seems its possible for `wp_generate_uuid4` to generate duplicates in some rare instances, but given we are only storing these secrets between magic link creation and login and at most until logout, I think it should be fine.

But let me know if you think we should use something else here!

Reply:
With a rotating secret, I also think this is more than enough entropy.